### ISSUE TITLE
Pre-trained model usage

### ISSUE THREAD

**Issue by DinoTheDinosaur** [yongxuUSTC/sednn#24]
Hello!
I've discovered that you've shared a pre-trained model on TIMIT dataset, but is it possible to run it on own data?
I've managed to reuse the trained model, generated in the process of running _sednn/mixture2clean_dnn/runme.sh_. I trained the model on the subsample of LibriSpeech dataset, but I'm not too happy with the quality of enhancing. When I tried to substitute LibriSpeech model with your TIMIT model, I couldn't run the decoding due to the input shapes mismatch.
So, my question is, how can I get to work the pre-trained TIMIT model on my own data in python? My assumpion is that default input data format differs in the _clean2clean_verify_ and _mixture2clean_dnn_ subdirectories. In that case where can I retrieve the information (scripts in the subdirectory _sednn/clean2clean_verify/_; papers)?

Best regards,
Yakovenko Olga

**Comment by DinoTheDinosaur** [comment]
> check the format of data is .wav and both the speech and the noise sampling rate is 16 kHz.
Yes, the data format is correct! The model works well on the test samples, but poorly on the real-life samples. That's why I also wanted to test your TIMIT model on the real-life samples - to compare the results.


> We are concatenating 11 frames as the input to the model. Please also do not forget to subtract the mean and divide the standard value of the input.
Ok, I see! I will try it out with these parametres.
Considering normalisation of the input - if I understand correctly, this functionality is covered by a scaler ('_scaler.p_' in _mixture2clean_dnn_ or '_tr_norm.pickle_' in _clean2clean_verify_). I tried to load the scaler with python 3 pickle, but I've encountered an error:
```
>>> AudioEnhancerTIMIT()
---------------------------------------------------------------------------
~/tools/sednn/mixture2clean_dnn/enhance_audio.py in __init__(self)
     56         def __init__(self):
     57                 self.model = load_model('model/sednn_keras_logMag_Relu2048layer1_1outFr_7inFr_dp0.2_weights.75-0.00.hdf5')
---> 58                 self.scaler = pickle.load(open('model/tr_norm.pickle', 'rb'))
     59                 self.n_window = 512
     60                 self.n_overlap = 256

UnicodeDecodeError: 'ascii' codec can't decode byte 0xe3 in position 0: ordinal not in range(128)
```
I heard there are sometimes issues with python's pickle files transfer between versions of python, maybe you have a newer version of scaler you can upload? Unfortunately, I have no access to the TIMIT database to calculate the scaler myself

**Comment by qiuqiangkong** [comment]
We did not keep the trained model on our disk, as we thought it would be easy to run the code to get the model. If rerun the code on TIMIT dataset successful, on the test dataset a PESQ of 2.08 +- 0.24 can be obtained. Then you could move on to the real-life samples to see how it perform. As that you do not have access to TIMIT dataset, we will consider to release a latest trained model and scalar who do not have access to TIMIT dataset.


The code is developed using python 2. You may need import _pickle as cPickle or trying something fancy like: cPickle.load(f, encoding='latin1'). I found these solutions on Google.


Best wishes,


Qiuqiang

________________________________

**Comment by DinoTheDinosaur** [comment]
> We did not keep the trained model on our disk, as we thought it would be easy to run the code to get the model. 

Yes, I understand! It is easy, thank you for making it that way :)

> As that you do not have access to TIMIT dataset, we will consider to release a latest trained model and scalar who do not have access to TIMIT dataset.

Ok, that would be great!

> The code is developed using python 2. You may need import _pickle as cPickle or trying something fancy like: cPickle.load(f, encoding='latin1').

Thank you, the second solution does work! Although later in the code I'm still met with the following error `ValueError: Error when checking input: expected input_1 to have 2 dimensions, but got array with shape (249, 11, 257)`. It looks like there are some differences in the training procedure between _clean2clean_verify_ and _mixture2clean_dnn_.


FIY, me and my team from Novosibirsk State University plan on making an open-source python package for noise supression in audio based on your repository (https://github.com/nsu-ai-team/noise_supression). There I already wrote a pipeline for single-file enhancement and I plan to upload LibriSpeech model (PESQ=1.82 +- 0.24), trained using your repo. Is it alright with you? Of course, I will add links and references to your repo.

**Comment by qiuqiangkong** [comment]
Hi, many thanks! That is absolutely fine!


Best wishes,


Qiuqiang

________________________________

**Comment by yongxuUSTC** [comment]
Hi, Yes, It is also fine for me.

Best regards,
yong

**Comment by DinoTheDinosaur** [comment]
So, I guess, there isn't a way to recover the parametres for the pretrained TIMIT model nowadays? If this is the case, it would be better to close the issue and wait for the uploads.
